src/MainWindow.cpp

```diff
@@ -337,8 +347,14 @@ MainWindow::hasActiveUser()
 {
         QSettings settings;
 
-        return settings.contains("auth/access_token") && settings.contains("auth/home_server") &&
-               settings.contains("auth/user_id");
+        UserSettings us;
```

Comment:
Why read the settings twice?

Reply:
Please elaborate.

Reply:
UserSettings uses QSettings. I don't see, why you would need to parse the same settings file twice. UserSettings should just provide access_token, homeserver, etc

Reply:
Gotcha.

Reply:
Could the part that goes:
```
        return settings.contains("auth/access_token") && settings.contains("auth/home_server") &&
               settings.contains("auth/user_id");
```
be switched to something like:
```
        return userSettings_->userId() != "" && ...
```